src/cascade/dismod/serialize.py

```diff
@@ -506,28 +527,52 @@ def smooth_id_func(smooth):
     return smooth_table, grid_table, smooth_id_func
 
 
-def make_rate_table(context, smooth_id_func):
+def make_rate_and_nslist_tables(context, smooth_id_func, location_to_node_func):
     rate_to_id = {}
-    rows = []
+    rate_rows = []
+    nslist_rows = []
+    nspairs_rows = []
     for rate_id, rate in enumerate(context.rates):
-        if len(rate.child_smoothings) > 1:
-            raise NotImplementedError("Multiple child smoothings not supported yet")
-
-        rows.append(
+        nslist_id = np.NaN
+        child_smooth_id = np.NaN
+        if rate.child_smoothings:
```

Comment:
What happens if the rate doesn't have any child smoothings? Every rate needs some smoothing. Does that mean it shares the parent smoothing, and should I see some registration of that parent smoothing as being associated with that rate? I'm concerned this might miss a case.

Reply:
From Brad's docs: "If child_smooth_id is not null, child_nslist_list must be null. If both are null, there are no random effects . for this rate."

It seems like if the modeler supplies no random effects smoothings, which is the else case here, they should get the "no random effects for this rate" case? Or am I misunderstanding something?